tests/java/org/python/core/ReflectedArgsTest.java

```diff
@@ -1,53 +1,167 @@
+
         public static void bar(int i, int[] j, int... rest) {}
+
+        public static void baz(Integer i, Integer[] j, Integer... rest) {}
+    }
+
+    /** {@code static void foo(Object... args)} */
+    private static final ReflectedArgs FOO_SIGNATURE =
+            new ReflectedArgs(null, new Class<?>[] {Object[].class}, Demo.class, true, true);
+
+    private static void fooCheck(Object[] args, Object... expected) {
+        // Expect one element that is an array of the expected values
+        assertEquals(1, args.length);
+        Object[] v = (Object[]) args[0];
+        assertEquals(expected.length, v.length);
+        for (int k = 0; k < expected.length; k++) {
+            assertEquals(expected[k], v[k]);
+        }
+    }
+
+    /** {@code static void bar(int i, int[] j, int... rest)} */
+    private static final ReflectedArgs BAR_SIGNATURE = new ReflectedArgs(null,
+            new Class<?>[] {int.class, int[].class, int[].class}, Demo.class, true, true);
+
+    private static void barCheck(Object[] args, int i, int[] j, int... rest) {
+        // Expect three elements conformant to the method arguments
+        assertEquals(3, args.length);
+        assertEquals(i, (int) args[0]);
+        assertArrayEquals(j, (int[]) args[1]);
+        assertArrayEquals(rest, (int[]) args[2]);
+    }
+
+    /** {@code static void baz(Integer i, Integer[] j, Integer... rest)} */
+    private static final ReflectedArgs BAZ_SIGNATURE = new ReflectedArgs(null,
+            new Class<?>[] {Integer.class, Integer[].class, Integer[].class}, Demo.class, true,
+            true);
+
+    private static void bazCheck(Object[] args, Integer i, Integer[] j, Integer... rest) {
+        // Expect three elements conformant to the method arguments
+        assertEquals(3, args.length);
+        assertEquals(i, (Integer) args[0]);
+        assertArrayEquals(j, (Integer[]) args[1]);
+        assertArrayEquals(rest, (Integer[]) args[2]);
     }
 
     /*
-     * If a function accepts Object Varargs (or Collection/array etc), the arguments
-     * should be boxed even if the final arg is a PySequenceList if the number of arguments
-     * not equal to the expected number of arguments.
+     * If a function accepts Object Varargs (or Collection/array etc), the arguments should be boxed
+     * even if the final arg is a PySequenceList if the number of arguments not equal to the
+     * expected number of arguments.
      *
-     * eg foo(1, [1, 2, 3]) should be valid for a call to
-     * void foo(int i, int[] j, int... rest)
-     * but should be boxed as foo(1, [1, 2, 3], [])
+     * eg foo(1, [1, 2, 3]) should be valid for a call to void foo(int i, int[] j, int... rest) but
+     * should be boxed as foo(1, [1, 2, 3], [])
      */
 
+    /**
+     * Calling {@code foo(Object... args)} as Python {@code foo(3, 4, ["bar"])} calls Java
+     * {@code foo(3, 4, strlist})} where {@code strlist} is the representation of Python
```

Comment:
looks like a mismatched `}` after `strlist`

Reply:
And here I wrote `{@code foo(new Object[] {3, 4, strlist})}` first (or something worse) then having reconsidered I didn't delete enough of it.